src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariants.java

```diff
@@ -0,0 +1,1244 @@
+        GenotypesContext newGC = GATKVariantContextUtils.updatePLsAndAD(sub, vc);
+
+        // since the VC has been subset (either by sample or allele), we need to strip out the MLE tags
+        builder.rmAttribute(GATKVCFConstants.MLE_ALLELE_COUNT_KEY);
+        builder.rmAttribute(GATKVCFConstants.MLE_ALLELE_FREQUENCY_KEY);
+
+        // Remove a fraction of the genotypes if needed
+        if (fractionGenotypes > 0) {
+            final ArrayList<Genotype> genotypes = new ArrayList<>();
+            for (final Genotype genotype : newGC) {
+                //Set genotype to no call if it falls in the fraction.
+                if (fractionGenotypes > 0 && randomGenotypes.nextDouble() < fractionGenotypes) {
+                    genotypes.add(new GenotypeBuilder(genotype).alleles(diploidNoCallAlleles).noGQ().make());
+                }
+                else {
+                    genotypes.add(genotype);
+                }
+            }
+            newGC = GenotypesContext.create(genotypes);
+        }
+
+        builder.genotypes(newGC);
+
+        addAnnotations(builder, vc, sub.getSampleNames());
+
+        final VariantContext subset = builder.make();
+
+        return preserveAlleles? subset : GATKVariantContextUtils.trimAlleles(subset,true,true);
+    }
+
+    /**
+     * If --setFilteredGtToNocall, set filtered genotypes to no-call
+     *
+     * @param vc the VariantContext record to set filtered genotypes to no-call
+     * @param filteredGenotypesToNocall  set filtered genotypes to non-call?
+     * @return the VariantContext with no-call genotypes if the sample was filtered
+     */
+    private VariantContext setFilteredGenotypeToNocall(final VariantContext vc, final boolean filteredGenotypesToNocall) {
+
+        if (!filteredGenotypesToNocall) {
+            return vc;
+        }
+
+        final VariantContextBuilder builder = new VariantContextBuilder(vc);
+        final GenotypesContext genotypes = GenotypesContext.create(vc.getGenotypes().size());
+
+        for (final Genotype g : vc.getGenotypes()) {
+            if (g.isCalled() && g.isFiltered()) {
+                genotypes.add(new GenotypeBuilder(g).alleles(diploidNoCallAlleles).make());
+            }
+            else {
+                genotypes.add(g);
+            }
+        }
+
+        return builder.genotypes(genotypes).make();
+    }
+
+    /*
+     * Add annotations to the new VC
+     *
+     * @param builder     the new VC to annotate
+     * @param originalVC  the original VC
+     * @param selectedSampleNames the post-selection list of sample names
+     */
+    private void addAnnotations (final VariantContextBuilder builder, final VariantContext originalVC, final Set<String> selectedSampleNames) {
+        // @TODO: this TODO was in GATK - is this still the case ?
+        if (fullyDecode) {
+            return; // TODO -- annotations are broken with fully decoded data
+        }
```

Comment:
I don't think you are ever actually fully decoding the records, even if `fullyDecode` is true, unless I've missed something (see my comment above).